python/lsst/verify/tasks/testUtils.py

```diff
@@ -21,17 +21,90 @@
 
 __all__ = ["MetadataMetricTestCase", "ApdbMetricTestCase"]
 
+import abc
+
 import unittest.mock
 from unittest.mock import patch
 
+import lsst.utils.tests
 from lsst.daf.base import PropertySet
 from lsst.dax.apdb import ApdbConfig
 
-from lsst.verify.gen2tasks.testUtils import MetricTaskTestCase
+import lsst.verify.gen2tasks.testUtils as gen2Utils
 from lsst.verify.tasks import MetricComputationError
 
 
-class MetadataMetricTestCase(MetricTaskTestCase):
+class MetricTaskTestCase(lsst.utils.tests.TestCase, metaclass=abc.ABCMeta):
+    """Unit test base class for tests of `tasks.MetricTask`.
+
+    This class provides tests of the generic ``MetricTask`` API. Subclasses
+    must override `taskFactory`, and may add extra tests for class-specific
+    functionality. If subclasses override `setUp`, they must call
+    `MetricTaskTestCase.setUp`.
+    """
+    @classmethod
```

Comment:
Just naming the class is too inflexible, because the task may need specific constructor arguments (for example, `ApdbMetricTask` needs a custom config if we want to run it without a real database, such as in a unit-test environment). A factory is much more appropriate here, even if it is unpythonic.

Reply:
This feels like too many levels of indirection for what you are actually trying to do. setUp is a two line function, why not have children classes just implement what they need in this method? That is basically how TestCase already works.

Reply:
Because of weirdness in how `unittest`/`pytest` actually runs these tests, that _doesn't_ work -- `self.task` needs to be assigned by the generic class, or generic tests won't see it. :man_shrugging: Believe me, I did try alternatives before picking this architecture.